lnbits/core/crud.py

```diff
@@ -396,12 +373,18 @@ async def create_payment(
     # previous_payment = await get_wallet_payment(wallet_id, payment_hash, conn=conn)
     # assert previous_payment is None, "Payment already exists"
 
+    try:
+        invoice = bolt11.decode(payment_request)
+        expiration_date = datetime.datetime.fromtimestamp(invoice.date + invoice.expiry)
+    except:
+        expiration_date = datetime.datetime.now() + datetime.timedelta(days=31)
```

Comment:
In case of an error, default to 31 days.

Reply:
31 days? strikes me as very long time for an invoice expiry. i would go for 1 day.
or maybe even move it to an env variable?

Reply:
31 days is the maximum expiry for bolt11 invoices afaik